vsphere/datadog_checks/vsphere/cache.py

```diff
@@ -88,3 +90,9 @@ def set_mor_data(self, mor, mor_data):
         if mor_type not in self._content:
             self._content[mor_type] = {}
         self._content[mor_type][mor] = mor_data
+
+    def iter_all(self, resource_type):
+        data = self._content.get(resource_type)
+        if data:
+            for mor, props in iteritems(data):
+                yield mor, props
```

Comment:
Hum, why not just return the mors for the resource and let the caller iterate ?

```python
    def get_resource_mors(self, resource_type):
        return self._content.get(resource_type, {})
```

Generator are sometimes useful, but IMO it's better to avoid them when possible, it makes debugging/troubleshooting easier.

Reply:
Yeah I wasn't happy either, as I wanted to use an iterator. I've updated this code by removing this extra function, and changing the `get_mors` function to return a iterator on keys.